signer/src/network/libp2p/network.rs

```diff
@@ -505,4 +527,184 @@ mod tests {
+            "ab0893ecf683dc188c3fb219dd6489dc304bb5babb8151a41245a70e60cb7258",
+        )
+        .unwrap();
+        // PeerId = 16Uiu2HAkuyB8ECXxACm8hzQj4vZ2iWrYMF3xcKNf1oJJ1NuQEMvQ
+        let key2 = PrivateKey::from_str(
+            "0dd4077c8bcec09c803f9ba23a0f5b56eba75769b2d1b96a33b579dbbe5055ce",
+        )
+        .unwrap();
+        // PeerId = 16Uiu2HAkv4DBE9f9eg53RoRYsfuzJXdoAvRU91gb6oii5pTseo1j
+        let key3 = PrivateKey::from_str(
+            "bdbb219e045b12c12d99c86afd83764ea67a9e3c0127c0298d7c57b3597a4645",
+        )
+        .unwrap();
+
+        // Create the context for signer1.
+        let context1 = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.private_key = key1;
+            })
+            .build();
+        // Add key2 to the known signers for signer1.
+        context1
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key2));
+
+        // Create the context for signer2.
+        let context2 = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.private_key = key2;
+            })
+            .build();
+        // Add key1 to the known signers for signer2.
+        context2
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key1));
+
+        // Add key2 to the known signers for signer1.
+        context1
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key2));
+
+        // Create the context for signer2.
+        let context3 = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.private_key = key3;
+            })
+            .build();
+        // Add key1 and key2 to the known signers for signer 3. This simulates
+        // what an adversary signer might do, i.e. it would want to join with
+        // the known signers.
+        context3
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key1));
+        context3
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key2));
+
+        let swarm1_addr = Multiaddr::random_memory();
+        let swarm2_addr = Multiaddr::random_memory();
+        let swarm3_addr = Multiaddr::random_memory();
+
+        // Create the two trusted swarms.
+        let mut swarm1 = SignerSwarmBuilder::new(&key1)
+            .enable_memory_transport(true)
+            .add_listen_endpoint(swarm1_addr.clone())
+            .add_seed_addrs(&[swarm2_addr.clone(), swarm3_addr.clone()])
```

Comment:
Should we have only swarm2 here (same below)? Or we need swarm3 to be in the seed addr as a worst case scenario (like signers using a wrong config for the seed addr)?

Reply:
I was wondering why we have `swarm3_addr` in the seed addr of swarm 1: if it's an adversarial it shouldn't be there, right?

Reply:
Oh, yeah it's good it's there just to simulate full connectivity (like, giving them every possibility to connect) and that messages still are rejected/not received by the adversarial node.